codec/json_test.go

```diff
@@ -0,0 +1,28 @@
+package codec
+
+import (
+	"testing"
+)
+
+func TestJsonMapKeysAsStringCanonical(t *testing.T) {
+	handle := new(JsonHandle)
+	handle.MapKeyAsString = true
+	handle.Canonical = true
+
+	var b []byte
+	enc := NewEncoderBytes(&b, handle)
+
+	mapType := map[interface{}]interface{}{
+		2:  "interface{} key",
+		11: false,
+	}
+	enc.MustEncode(mapType)
+
+	// We expect "11" to be encoded before "2" because its string representation
+	// is lexicographically less than "2".
+	expected := `{"11":false,"2":"interface{} key"}`
```

Comment:
Yes, that's worth noting. As far as I can tell, this is how the upstream repo also implements canonical JSON ordering.

I prefer to think of the ordering as deterministic instead of canonical, so the exact order doesn't necessarily matter as long as it's consistent.